osslsigncode.c

```diff
@@ -4229,12 +4229,7 @@ static int read_crypto_params(GLOBAL_OPTIONS *options)
 
 #if !defined(OPENSSL_NO_ENGINE) || OPENSSL_VERSION_NUMBER>=0x30000000L
     /* Security token */
-#ifndef OPENSSL_NO_ENGINE
-    /* PKCS#11 'dynamic' engine */
```

Comment:
A simple change to the logic condition should be sufficient:
```
 #ifndef OPENSSL_NO_ENGINE
     /* PKCS#11 'dynamic' engine */
-    } else if (options->p11engine && !engine_load(options)) {
+    } else if (options->p11engine) {
+        if (!engine_load(options))
             goto out;
 #endif /* OPENSSL_NO_ENGINE */
```

Reply:
Yep, it works that way